internal/driver/metadata.go

```diff
@@ -137,19 +137,22 @@ func getDataFormat(d *usbdevice.Device) (interface{}, error) {
 	}
 	result.Quantization = quant
 	intervalCount := 0
-	var intervals []v4l2.Fract
+	var frameRates []v4l2.Fract
 	for {
 		fd := d.Fd()
 		index := uint32(intervalCount)
 		encoding := pixFmt.PixelFormat
 		if interval, err := v4l2.GetFormatFrameInterval(fd, index, encoding, pixFmt.Width, pixFmt.Height); err == nil {
 			intervalCount += 1
-			intervals = append(intervals, interval.Interval.Max)
+			frameRates = append(frameRates, v4l2.Fract{
```

Comment:
I would add a small comment in the couple of places that you do the swapping of num/denom with denom/num such that others looking at the code do not think it is a typo.